crates/primitives/src/trie/proofs.rs

```diff
@@ -83,4 +123,20 @@ impl StorageProof {
     pub fn set_proof(&mut self, proof: Vec<Bytes>) {
         self.proof = proof;
     }
+
+    /// Verify the proof against the provided storage root.
+    pub fn verify(&self, root: B256) -> Result<(), ProofVerificationError> {
+        let expected =
+            if self.value.is_zero() { None } else { Some(encode_fixed_size(&self.value).to_vec()) };
```

Comment:
```suggestion
            // If the storage `value` is Zero then it's no longer part of the trie
            if self.value.is_zero() { None } else { Some(encode_fixed_size(&self.value).to_vec()) };
```

this could also probably be `self.value.encode()` given `U256: RlpEncodable` seems like footgun